shared/message/support/FieldDescription.proto

```diff
@@ -22,42 +22,76 @@ package message.support;
+ */
 message FieldDescription {
+    /// The cross-sectional shape of the goalposts
     enum GoalpostType {
         RECTANGLE = 0;
         CIRCLE    = 1;
     }
-
+    /// Measurements of the field which are loaded from config file. In metres
     message FieldDimensions {
-        double       line_width             = 1;
-        double       mark_width             = 2;
-        double       field_length           = 3;
-        double       field_width            = 4;
-        GoalpostType goalpost_type          = 5;
-        double       goal_depth             = 6;
-        double       goal_width             = 7;
-        double       goal_area_length       = 8;
-        double       goal_area_width        = 9;
-        double       goal_crossbar_height   = 10;
-        double       goalpost_width         = 11;
-        double       goalpost_depth         = 12;
-        double       goal_crossbar_width    = 13;
-        double       goal_crossbar_depth    = 14;
-        double       goal_net_height        = 15;
-        double       penalty_mark_distance  = 16;
-        double       center_circle_diameter = 17;
-        double       border_strip_min_width = 18;
-    }
+        /// The rules say that the lines must be the same width
+        double line_width = 1;
+        /// The 'touch line' length
+        double field_length = 2;
+        /// The 'goal line' length
+        double field_width = 3;
+        /// Cross-sectional shape of goal-posts
+        GoalpostType goalpost_type = 4;
+        /// Distance behind the keeper to the net
+        double goal_depth = 5;
+        /// Distance between inner edges of goal posts
+        double goal_width = 6;
+        /// 'Goal area' is equivalent to 6-yard box in soccer. Length is front to back, if you standing in the middle
+        /// of the field facing goals
+        double goal_area_length = 7;
+        /// 'Goal area' is equivalent to 6-yard box in soccer. Width is left to right, if you're standing in the middle
+        /// of the field facing goals
+        double goal_area_width = 8;
+        /// Height to the bottom of the crossbar
+        double goal_crossbar_height = 9;
+
+        double goalpost_width = 10;
+
+        double goalpost_depth = 11;
+
+        double goal_crossbar_width = 12;
+
+        double goal_crossbar_depth = 13;
 
-    double ball_radius         = 1;
+        double goal_net_height = 14;
+        /// Measured from the goal line to the middle of the penalty mark
+        double penalty_mark_distance = 15;
+
+        double center_circle_diameter = 16;
+
+        double border_strip_min_width = 17;
+    }
+    /// @see Law 2
+    double ball_radius = 1;
+    /// Calculated as goal_crossbar_height + goal_crossbar_width
     double goalpost_top_height = 2;
+
     double penalty_robot_start = 3;
 
     // Coordinates of goalpost centers calculated from the FieldDimensions:
+
+    /// {x, y} of our left goalpost, calculated from goal_width and half_length
```

Comment:
What is `half_length`?

Reply:
My bad. That's a variable used in `SoccerConfig`, which is the only place this is used. I'll be more explicit, without having that logical dependency :+1: